ui/src/app/workflows/components/workflow-details/workflow-details.tsx

```diff
@@ -177,159 +177,56 @@ export class WorkflowDetails extends React.Component<RouteComponentProps<any>, W
         );
     }
 
+    private confirmAction(title: string): void {
+        if (!confirm(`Are you sure you want to ${title} this workflow?`)) {
+            return;
+        }
+        return;
+    }
+
+    private getHandleErrorFunction(title: string): () => void {
+        return () => {
+            this.appContext.apis.notifications.show({
+                content: `Unable to ${title} workflow`,
+                type: NotificationType.Error
+            });
+        };
+    }
+
+    private performAction(action: (name: string, namespace: string) => Promise<any>, title: string, redirect: string, ctx: ContextApis): void {
+        this.confirmAction(title);
+        action(this.props.match.params.name, this.props.match.params.namespace)
+            .then(() => ctx.navigation.goto(uiUrl(redirect)))
+            .catch(this.getHandleErrorFunction(title));
+    }
+
     private getItems(workflowPhase: NodePhase, ctx: any) {
-        const items = [
-            {
-                title: 'Retry',
-                iconClassName: 'fa fa-undo',
-                disabled: workflowPhase === undefined || !(workflowPhase === 'Failed' || workflowPhase === 'Error'),
-                action: () => this.retryWorkflow(ctx)
-            },
-            {
-                title: 'Resubmit',
-                iconClassName: 'fa fa-plus-circle ',
-                action: () => this.resubmitWorkflow(ctx)
-            },
-            {
-                title: 'Suspend',
-                iconClassName: 'fa fa-pause',
-                disabled: !Utils.isWorkflowRunning(this.state.workflow) || Utils.isWorkflowSuspended(this.state.workflow),
-                action: () => this.suspendWorkflow(ctx)
-            },
-            {
-                title: 'Resume',
-                iconClassName: 'fa fa-play',
-                disabled: !Utils.isWorkflowSuspended(this.state.workflow),
-                action: () => this.resumeWorkflow(ctx)
-            },
-            {
-                title: 'Stop',
-                iconClassName: 'fa fa-stop-circle',
-                disabled: !Utils.isWorkflowRunning(this.state.workflow),
-                action: () => this.stopWorkflow(ctx)
-            },
-            {
-                title: 'Terminate',
-                iconClassName: 'fa fa-times-circle',
-                disabled: !Utils.isWorkflowRunning(this.state.workflow),
-                action: () => this.terminateWorkflow(ctx)
-            },
-            {
-                title: 'Delete',
-                iconClassName: 'fa fa-trash',
-                action: () => this.deleteWorkflow(ctx)
-            }
-        ];
+        const actions: any = Actions.WorkflowActions;
+        const items = Object.keys(actions).map(actionName => {
+            const action = actions[actionName];
+            return {
+                title: action.title.charAt(0).toUpperCase() + action.title.slice(1),
```

Comment:
This seems a tad dirty. Is there a reason as to why `action.action` is lower case? Could it be capital case instead and if it needs to be lower case then we can do `action.action.lower()` or whatever the TS equivalent is

Reply:
yeah I wasn't wild about that either. Capital case is a good solution and `*.lower()` is certainly cleaner, I'll do that.

Reply:
It doesn't seem like this was changed?